state/state_test.go

```diff
@@ -121,6 +164,60 @@ func TestStateSaveLoad(t *testing.T) {
 	assert.True(state.Equals(loadedState),
 		fmt.Sprintf("expected state and its copy to be identical.\ngot: %v\nexpected: %v\n",
 			loadedState, state))
+
+	// the following assertions are just for additional assurance
+	assert.Equal(state.TimeoutCommit, loadedState.TimeoutCommit,
+		fmt.Sprintf("expected TimeoutCommit to be equal."+
+			"\ngot: %v\nexpected: %v\n", loadedState.TimeoutCommit, state.TimeoutCommit))
```

Comment:
[nit] the default `assert.Equal` will report information like this if the assertion fails so this doesn't need to provide a custom error string.